source/digits_hits/src/GateDoseActor.cc

```diff
@@ -379,29 +386,34 @@ void GateDoseActor::UserSteppingActionInVoxel(const int index, const G4Step* ste
 
       double cut = DBL_MAX;
       cut=1;
-      G4String material = step->GetPreStepPoint()->GetMaterial()->GetName();
-      double Energy = step->GetPreStepPoint()->GetKineticEnergy();
-      G4String PartName = step->GetTrack()->GetDefinition()->GetParticleName();
+      G4Material * material = step->GetPreStepPoint()->GetMaterial();
+      static G4Material * water = G4NistManager::Instance()->FindOrBuildMaterial("G4_WATER");
+      double energy = step->GetPreStepPoint()->GetKineticEnergy();
       double DEDX=0, DEDX_Water=0;
 
       // Dose to water: it could be possible to make this process more
       // generic by choosing any material in place of water
-      double Volume = mDoseToWaterImage.GetVoxelVolume();
-
-      // Other particles should be taken into account (Helium etc), but bug ? FIXME
-      if (PartName== "proton" || PartName== "e-" || PartName== "e+" || PartName== "deuteron") {
-        //if (PartName != "O16[0.0]" && PartName != "alpha" && PartName != "Be7[0.0]" && PartName != "C12[0.0]"){
-
-        DEDX = emcalc->ComputeTotalDEDX(Energy, PartName, material, cut);
-        DEDX_Water = emcalc->ComputeTotalDEDX(Energy, PartName, "G4_WATER", cut);
-        doseToWater=edep/density/Volume/gray*(DEDX_Water/1.)/(DEDX/(density*e_SI));
-        if (DEDX_Water == 0 || DEDX == 0) doseToWater = 0.0; // to avoid inf or NaN
+      double volume = mDoseToWaterImage.GetVoxelVolume();
+
+      // Get current particle
+      const G4ParticleDefinition * p = step->GetTrack()->GetParticleDefinition();
```

Comment:
Not sure ... I think it is a pointer to const and not a const pointer, meaning the object pointed to is const.